xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/UserMentionsFormatter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.mentions.DisplayStyle;
+import org.xwiki.mentions.MentionsFormatter;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.user.UserProperties;
+import org.xwiki.user.UserPropertiesResolver;
+import org.xwiki.user.UserReference;
+import org.xwiki.user.UserReferenceResolver;
+
+import static org.xwiki.mentions.DisplayStyle.FIRST_NAME;
+import static org.xwiki.mentions.DisplayStyle.LOGIN;
+
+/**
+ * User implementation of {@link MentionsFormatter}.
+ *
+ * @version $Id$
+ * @since 12.10RC1
+ */
+@Singleton
+@Named("user")
```

Comment:
You're using the "user" hint as a fallback at different places (in provider and in consumer if I'm correct), so IMO it deserves a public constant.

Reply:
`org.xwiki.mentions.MentionsConfiguration#USER_MENTION_TYPE` introduced.